lib/Mojolicious/Plugin/OpenAPI/Guides/OpenAPI3.pod

```diff
@@ -0,0 +1,392 @@
+    my $output = {pets => [{name => "kit-e-cat"}]};
+    $c->render(openapi => $output);
+  }
+
+  1;
+
+The input will be validated using
+L<Mojolicious::Plugin::OpenAPI/openapi.valid_input> while the output is
+validated through then L<openapi|Mojolicious::Plugin::OpenAPI/RENDERER>
+handler.
+
+=head2 Route names
+
+Routes will get its name from either L</x-mojo-name> or from L</operationId> if
+defined in the specification.
+
+The route name can also be used the other way around, to find already defined
+routes. This is especially useful for L<Mojolicious::Lite> apps.
+
+Note that if L<spec_route_name|Mojolicious::Plugin::OpenAPI/spec_route_name>
+then all the route names will have that value as prefix:
+
+  spec_route_name            = "my_cool_api"
+  operationId or x-mojo-name = "Foo"
+  Route name                 = "my_cool_api.Foo"
+
+You can also set "x-mojo-name" in the spec, instead of passing
+L<spec_route_name|Mojolicious::Plugin::OpenAPI/spec_route_name>
+to L<plugin()|Mojolicious::Plugin::OpenAPI/register>:
+
+  {
+    "openapi": "3.0.2",
+    "info": { "version": "1.0", "title": "Some awesome API" },
+    "x-mojo-name": "my_cool_api"
+  }
+
+=head2 Default response schema
+
+A default response definition will be added to the API spec, unless it's
+already defined. This schema will at least be used for invalid input (400 - Bad Request) and
+invalid output (500 - Internal Server Error), but can also be used in other cases.
+
+See L<Mojolicious::Plugin::OpenAPI/default_response_codes> and
+L<Mojolicious::Plugin::OpenAPI/default_response_name> for more details on how
+to configure these settings.
+
+The response schema will be added to your spec like this, unless already defined:
+
+  {
+    ...
+    "components": {
+      ...
+      "schemas": {
+        ...
+        "DefaultResponse": {
+          "type":     "object",
+          "required": ["errors"],
+          "properties": {
+            "errors": {
+              "type":  "array",
+              "items": {
+                "type":       "object",
+                "required":   ["message"],
+                "properties": {"message": {"type": "string"}, "path": {"type": "string"}}
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+The "errors" key will contain one element for all the invalid data, and not
+just the first one. The useful part for a client is mostly the "path", while
+the "message" is just to add some human readable debug information for why this
+request/response failed.
+
+=head2 Rendering binary data
+
+Rendering assets and binary data should be accomplished by using the standard
+L<Mojolicious> tools:
+
+  sub get_image {
+    my $c = shift->openapi->valid_input or return;
+    my $asset = Mojo::Asset::File->new(path => "image.jpeg");
+
+    $c->res->headers->content_type("image/jpeg");
+    $c->reply->asset($asset);
+  }
+
+=head1 OpenAPIv2 to OpenAPIv3 conversion
+
+Online and offline tools are available one example is
```

Comment:
This sentence reads a bit strange. Should it be "...are available. One example..."?

Reply:
agree strange indeed, tried with this

"Both online and offline tools are available. One example is of this is"